### ISSUE TITLE
cannot browse SMB shares in file manager

### ISSUE THREAD

**Issue by rkitover** [mpv-android/mpv-android#355]
I have my NAS mounted on my shield android tv device, but I see no way to access files on it from the app file manager. The file manager starts at `/storage/emulated/0` and clicking the `..` button does nothing. Changing the default file manager path in settings to my share, which is `/storage/NAS/rkitover` and restarting the app has no effect.

**Comment by sfan5** [comment]
On devices which provide the relevant API mpv only allows access to storage volumes.
I take it you're rooted and mounted it there by hand?

**Comment by rkitover** [comment]
No this is regular "attached network storage" from the android system, I can browse all of that stuff fine in file managers like total commander or solid explorer, and I can access it with Nova player. Currently I can launch mpv on a file from total commander, but it would be really nice if I could open my NAS directly in mpv.

**Comment by rkitover** [comment]
I can look into this, it may be just a matter of getting another permission, I've never done anything with android and need to learn more about it.

**Comment by sfan5** [comment]
I doubt that it's a permission issue, mpv-android uses this code to detect storage volumes it can directly access:
https://github.com/mpv-android/mpv-android/blob/21d8388e1c498af729057487b12d71056549d0b1/app/src/main/java/is/xyz/mpv/Utils.kt#L73-L97
The way we do that is not perfect however since Google doesn't like apps directly touching the filesystem and subsequently does not provide APIs for that.

Anyway here's an APK you can test: https://kitsunemimi.pw/tmp/mpv-android-2021-02-01.apk
Please try it out and show me a screenshot of the debugging output in "Settings > Advanced > Version information".

> No this is regular "attached network storage" from the android system

Then, what functionality or app did you use to mount it?

**Comment by rkitover** [comment]
> Then, what functionality or app did you use to mount it?

This is android tv 9 on my shield, I go to `Device Settings -> Storage -> Connect network storage` to connect my NAS.

I got the screenshots, it would be easier if you could just copy the text in the version information dialog:

![screenshot 1](https://lh3.googleusercontent.com/kV9e-edFLWFrHT9Zmhf8ecx0S-c4aVMoZh6bGaidfP78IX5GeyvRJ3t2q7b-j64k7xeBz5tLcSLz0kiSv7lVAuazUnbqqXW5QD9a28ns6cUHrcE0ZGMGDeLAhoXZO4joAZrUqooW6ecaPfOmVhf7oS3eVtjJPnjEmtKDoVkmfY-QjKtt58XUj7qOYFdXnsOpzQZmL4jUzM_tnrXu213nkPXKU1J0GaWR-K0DYncRICKg2wOfCLZMWCYzTQ2AWRg7J6-Gaye7sqw8Q3EjBJ3DMibUA8340oy1i-0dAoo21Sm6V0dzQT8DWKDaDqJjgzoNydMoawmyzvsPFAT7faxpvv89UF3Ol5f2cFI3LEQ0v9rIUzwtYUheRiSTYKVaHjmhuJ-0-k-e9QwPE1FN7EYRWRhBtL7Ln12f0yc6xCsg8_k3iry3L_Ywp0nQ7YvFWCbUMVpxtqts3tQg5bKVsSbuW9HwhGxL8xDdYkma3SzCiWWhFFQPBjzbapbfuVgdv8asC4aKKwM4fb5e3Lthe0e9esVZBbzDC9s3-V7aC7QOaYgWgKfdM8-HT7whdRINqs5TzEHEXL-EaoNASUZ3xyvnuaaOjYU3hxfx5gCLeAu_DDDjrIW04yleZFQ4XVNA8QlzqjttmStRrveCvli5R2MOzs7IY8m1Gk_uAjmQIswbQABXdzxIqLEsa0KtQp8A=w800-h533-no?authuser=0)

![screenshot 2](https://lh3.googleusercontent.com/-tPz2KbvgdZ9W8VlQylnv3JfEgsXo39ricHNlrHiH_LaI4PkZ-M0F-WYOnC4E9AqjhTjI7weR_F-5bN-SvMucj4pIuQTBZJw_yyc6yUqmvDlQuX6I0k67wg5Iui5ZtCMhEMzmBSmSmB3hoW6TM6q4eLPrCthE8x46N0QcKdBHKgRwRCT_qDZuk6kcN7W9YkHgBbIF0VhtHlrNBko9mWHCBNMVrBxcnRtvNC1k0gDcQ4HF8wY6keqMsokrsZSG0bklAplHpsndrGFAXsqYQvFBbIV_8gVp2iY_B0OWl4CMO1vh_OZwfJMczJTZDuapkYcDb7xIzefXdzOh0kVGeF_MGYJ2QfyABHxjFLzt8y9j_Ifo5W-8S0NaF5RsLin1ftbl8ouJQR_MNYVzx6x-Nn699SUt4YrvMvT0qlnBvvgjU1jBHxr5sQ9g1SVmvuQ2Wp7taGtbXQozlT4e-cFpdrzOy2cc_06WCkaCBidGLDoB1-3mQKa_IAiInNbI_WiuTuwf9Fvb5_ZyHpNmakR2UxbCRA7TQupWwSwP2lrrurwwwTd1J_fXX2YLeIBuuKYBJKp3hLdYDkAZ4U7uoxJyJKJB5BFJGolTjkCU1TxSxCdrOORmFrmDOCfLuNAnRkReQ7sx22d3RdlUdkcjoJ6-TqEDBv-5PemCQPfJvRtKiloVQLH4FBZEaUG5AgEdLnh=w800-h533-no?authuser=0)

![screenshot 3](https://lh3.googleusercontent.com/PutM6oQSh8XuZf-aIVOg6t97G1OWkNZ0G29YsxMrT-3yCzNK9ddASVw1mBwne8yo3Ob48zE2DlAyoGRgv6tb4HCEmNLwejDHTk86nJmJ5Ic_nsp4on_8RbAOYXBIIjWTiX5tqCgyygg2q9_iMmOMsnGZRjxTO7sy0t0N34KnG68h4DD98ATgywr8OHzs6F-J4WmPfoXGU-nHnEeWNfOF2v_a7UYslb5FGwrVYlAL78Jeawc0firIkaPxDiHu06S7T_R5Q50K29CsCOCW0uTI0QFp0lNes-39A7VonNGt2Ch36902Ekmm8IB038OTGN6nrbqJxM8TkPS0cncHNgxSQMx3Tj3X3azkqrW6_m0WvqSbFPD_LzhBpsM56LGWKeYOCmtjSF-wp5ADtjmNFYPH6403bmeQynNz94QuVtB4o-RDNHYnczjjwoitbsZWR6IbjwzbeI25HjZEo_2TjUQ1LHU23ncbpoEu2rUz1slrYfOMMuJNGWc489ZnpmNNTYmzXfFPWegPKOOk7fiWWlZJg6_nMAICF6k4E0J-9gvS31kvv8NJcHvMUZ_dXF59m8MiGhO-MWOJswnUazXp34e_Hkd73qB0_XOZOAwgvnxVi5OyHotnCzlSBT5FcXzPMXrHfbHZmF77iCFnBhKV9mZB6-rA_W6GLReOg5fHI0Tc1ccMCtm5rQ4HjzA0mUpu=w800-h533-no?authuser=0)

![screenshot 4](https://lh3.googleusercontent.com/8_YBnbbAyZ8bABWF0vX0pva4NJ5qP9NfQ5HuMopBKaIMY7SFiAwQgo4mvsNuHmkdBg_RAA8FVk-z6L85hw5wAj__wDMEydMJNSEeNiS87gZ5ts_j6P4_2-aRk4sSeSKYNx-a0NE6weaheS33f7M48EeF5MCvMsYt8y1LFGFj34mz2XLlG6fFZza_7exkKN2pqsXV6ce1It08WAd8FHlM55KEQxw2sZEBVChxAVSaAnspkuHyRpjsEM0RnGER8xuJYwUaQTKehnd31zCB4qoWARHkXAkL9Scx6qQUO-FcUPXFZGJ2XJZ_rLtceCmIpcgWmOzcv1aqzLF8RBxSMaFX3Amv_OvaK-U2mQwbvrixotYqUBMbaVYu_1mudu21YHK70j_hNBU0EdGdm-ZORi3lpn4L8TMzkcYVD9s_aFuHoOPFc98EkEeDntnlGx1wPiKvS99sMgNNFPbl0gWx3dJhKnrCJ5tWKs79k7N766l1nZdQ0og47ixuVCE5cwaMpbkiaaqvScR7pmBCLdLaixbC2NRnDxlOtFao-m_J-4hv_L0y-wKlaioplD1jiXrc-aT_Vry3_L0kOaraxBu6BHwjl5FAGtUbOXGcUQkH9v4wxBthPJwHQ_VWokbhVVRT6czm6I7X5gnfG2-2RMjIWVnO-IxItLA4Tk8cLFoQnIqDpVBHljZL7L3LuUGDze9x=w800-h533-no?authuser=0)

![screenshot 5](https://lh3.googleusercontent.com/5RTqD1JQ4a2iR1wM3cz1ONX9kG8lJpYYf2v7fe2f_4AOM5RlJ_nr2mQR-j8UiqRZO47JdJZfFiTBskaYAWGjRGNJrpRi4p2geLpQvaEOz3B20nCzKgHB2GRV6XXx0yoVr-59g5k8IThIuZbsqZorE0HtntZes_cmlM1b59vc9Ei_z3bmi9ULAF9J5JsrZb7FJYV0hOxT47JXZmxQDbP0HWVCXZBle0tH7ZIr7otUglvJpqgoH14rYwmsWuU2kvRe-3n9KkzKapO2azsPpnYX_W_p3B5W-nWJ4xCDt_tC1mOdDYmi23hGdx5aKu_YMR0nRaSnDlVOKpK_hTz5iL4gcl9q5kzWZPPJRmfodFXkl-annZ7-GbdLr5liwPXAVbE70BKu3ma87YdDlUzUkg_CRpiHN4zQp6XkYSZSZRan54yo9bdkzw-qgluaIvLTkE5da5WGFOQmPtiR5QkA1fvgtdfw9AoMQM_YxBB0xym3wn5LNwNEvoplyswpS6TqcNwEYQsrVwSCREFvhUp9_k9AsLlpVr1-zqFaVH1EOFvbxgJGUCiI_EZhHPlbbxy7Ri7Zocp-JEcNq7OInl0Y0fth0aOs2U5_SOG4LFTGys6nH8WF-7KG2PsKF0HX-o97IfJIGdyJVuovxcrF3ISKkYknvixp3soIQWXvSMywK3DC8R9IqyllOQRpluUXSki-=w800-h533-no?authuser=0)

![screenshot 6](https://lh3.googleusercontent.com/n4i61enPO4Zg18b-GGpRf0961Y8Xrd4YF2-4Yrn0WxKZp6hTHVJXYKlmioo7DO3eVDp-uynV9oQ2XTjF2w6H0R9YzflDZ8gQNeQQIcJzxTFkU0UmLPQOCDASgsp0FFTbLLAGDxNTMpC-NTXRhZWhcl2t5oF74fUn62PXx34O4KTngdpfbdNC-pBSp7wJrznWyBrDGZ---G1mCGOOAEQNu5CtTQ90NHKJZqlthB2p7R35bGKGul-EezTsk7IDvDEzKYMoLxRjGrXwLLNc-JfvGM6ZDLYFpMpbqVtWQ85H1mMdH8mMlpnhseWBsq78U5BWvpm-9agDKvBumWM4ZWwh88sYycHv2DM6xAWq1mOzmuC-RLV3ieb2xPx9vecvzGsQ7OIt_z5C1kVSbt0HUjI_fWV0YGeFtUVsg1rhq_V6hgL_IXFukG0wJUY2MOxz5PInZR7Eki0kcIBm9Uep37VjXzIZhSAvvsM46fPLspQHyqHq18t1jEiDnlMIPQqrGok1i7Hydj2LoeK1s7cXfU18XL2Gao1K1qjr8GxAdEX2yh8j_JAqikdgr--nUa2NrwTwfRVCnA-DjSFHkIeXgnP7c217uFQieIYs40SDAPaVI6YV2P6WSsYcLPqph02eszh_6AP8yLMKcYbWeJHPtcfnJ0BJrmOAeJPbE3TE-jmKycl7uSzt-C6BgUyyY1T9=w800-h533-no?authuser=0)

![screenshot 7](https://lh3.googleusercontent.com/faCZ4KWieHm0ngAzPqm0C5bGPXkCifwFXNr2NnpMPjHzz8LNdrbpaFiju4uf0jTF07RgN2P5bg4cZPQgDm78KVe72s_4hfa96p_EUhMBWuIz_gxDGz2tMWffqC86h4o0jLRgbzjMy3DbEWsZjYqAQJBxm2jnN1qDlKE0NnQQtts-YdUWoJcHE_40u3ThtTkHdFCB7_W_aV9joL5JE0QSBi1ctRSDPhLHT2zZpXlN-AwOFSHB_kfNKBK-UoIK9hp5LSL1Tr08T5F2m1bqIDDjJgSDxYVOo_hjO2l9Pln3vDaMqDLL2f4KKzsleP53fbkAEhzXAm7JWGDEKfwIo6dlqCtmyd5RsqE2e7x5RCsuOQO16pejwQQB3gMpEcLCesbqU1AqKK_0MDnWQEWB2sxnUJT5ZV89HIs1c6kR8lN3ZmsuZBguWAR0ZmnNIoJjRpFVlJkBDVTc1MrscmtbB0LRT4co0j9CBaE_0ifZGyzHy2rAP5ELpU5_Dcxc3nak_nxpmB9pfheGPSdBAKEeXym_sbk_WW3dQN3qRw-PjWwEXW7awKDJqciLzy48UAXXb_FeQW5IuVu9VQst8V4vs1N4An_hd8xR738K63AQpCUuxEKagbtyawSP6L2GbBc5YrtVTAcuLMTI82MS5kviKnBsIf8G4oWfznbWjvRBGQky4RVc3NQE4wlm7XdDulNS=w800-h533-no?authuser=0)

![screenshot 8](https://lh3.googleusercontent.com/ICP72GKFo7REj4E01OChDH--PuapjA6LPcE0Psa8E1CdfbvlhjrUERZoLC8rMPj0uxYO_dgCbc7Eg_nKlx1IUNnC-28vd8guyNnXFlClF3BUavUOa6W5Pj6KsKeoTGMrkDlEFm9OoJbFh6g2lzTqCJEDk6-ctfhNQiZwJeD6KHJVZXN1i53G_z2t3Wzwki8P_-38uH8Ix0Cde-SfXpscuI8CoaxYN9xi7LY-x_smcOGzovOzo1Ob033ki-gXBc0uWgcxNWTgQc489t3BF3nd7By78ovSe7JvBT2WT-kc1TCghMdEPFiUZnTb4bA-8yY4yQYaitj1BoUrl-Un6PFgovkzE2C8aU9ycuZ2j2z3VXnGXeHT59OtdhXnY7veg06rO8BYctYRKzItDM5-HeIXQeW9ok2y37BnrAHCfMRXDRk685PM7g9eX31rm0LfIQve-RAFkU3-K671793r_om1ClAsgdVgp3g2sZP5FNhX60Uu1aWXSReoSO58afUUe0zyYXgOisNz9PTcFhOxSFP5GPo0Ap6gtXR9Rb0RojClwf1ismeXcnEutmIrK_AxeJyaOGoh2oUZFTvp9UJumw9G7y2Fjo3WuxuYqQ32w2RFPDQi3QiVykm7ZGtamPtRcAJkSwtgyLFET0r70hr0K1CFDS8yIQc2_dITxbWXjQ5Ld0ZO-5C3PJzh3xeHSO6e=w800-h533-no?authuser=0)

**Comment by sfan5** [comment]
Those images don't load here, not sure how you uploaded those but you can also just drag them into the comment field.

**Comment by rkitover** [comment]
Sorry I did that wrong, let me try again:

![IMG_20210202_0438132](https://user-images.githubusercontent.com/77611/107071722-0e488880-67dd-11eb-9d8b-787556bd012f.jpg)

![IMG_20210202_0438351](https://user-images.githubusercontent.com/77611/107071766-1d2f3b00-67dd-11eb-8f13-370247f16bfe.jpg)

![IMG_20210202_0438541](https://user-images.githubusercontent.com/77611/107071773-21f3ef00-67dd-11eb-8069-cbd366087e6e.jpg)

![IMG_20210202_0439153](https://user-images.githubusercontent.com/77611/107071785-27513980-67dd-11eb-9e5d-fd3d9395a421.jpg)

![IMG_20210202_0439274](https://user-images.githubusercontent.com/77611/107071810-3506bf00-67dd-11eb-933e-cf81ab600dff.jpg)

![IMG_20210202_0439493](https://user-images.githubusercontent.com/77611/107071839-3df79080-67dd-11eb-9d2b-6ad863bade9c.jpg)

![IMG_20210202_0440126](https://user-images.githubusercontent.com/77611/107071865-45b73500-67dd-11eb-8631-0befe111d198.jpg)

![IMG_20210202_0440230](https://user-images.githubusercontent.com/77611/107071884-4bad1600-67dd-11eb-8d80-ba654188a244.jpg)

the drag and drop method is definitely simpler than what I was doing before.

**Comment by sfan5** [comment]
Thanks for testing, it looks like it's finding all the right mounts.

If you use the sd card icon in the mainmenu, do they show up there? do they work?
If so I can push this fix right into the repository.

**Comment by rkitover** [comment]
I just tried using the sdcard icon in that build, I was able to select the SMB mount, browse to my anime video and play it. This actually works better than launching mpv from total commander for some reason, I can switch audio and subtitle tracks faster and it shows their full name.
